cmd/istioctl/register.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+	"k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"istio.io/pilot/platform/kube"
+)
+
+type namedPort struct {
+	Port int32
+	Name string
+}
+
+func (p *namedPort) String() string {
+	return fmt.Sprintf("%s:%d", p.Name, p.Port)
+}
+
+var (
+	// For most common ports allow the protocol to be guessed
+	portsToName = map[int32]string{
+		80:   "http",
+		443:  "https",
+		8000: "http",
```

Comment:
it's just a shortcut, see the examples (ie you can do   mysql:1521  grpc:8079 etc)